Core/Handlers/MessageHandler.php

```diff
@@ -381,9 +393,15 @@ public function onHandleException(
                 $this->deferExchangeMessage($retryExchangeEnvelope, $this->retryURI);
             } // If it's an exchange that is failing and it should not be retried later
             else {
```

Comment:
This condition instead of an else it should be an else if and in this case I think it could be simplified like this
```
elseif ($context->get(Context:: CONTEXT_CALLBACK) === true &&
$context->get(Context::CONTEXT_CALLBACK_METHOD) != '')
```
To do this it requires to add in the code a line 
```
$context = $exchangeBackup->getIn()->getContext();
```
I suggest to add this line of code after the dispatch event code and before the first if